pkg/controller/machine/machine_controller_phases.go

```diff
@@ -251,3 +252,25 @@ func (r *ReconcileMachine) reconcileInfrastructure(ctx context.Context, m *v1alp
 	m.Status.InfrastructureReady = true
 	return nil
 }
+
+// reconcileClusterAnnotations reconciles the annotations on the Cluster associated with Machines.
+// TODO(vincepri): Move to cluster controller once the proposal merges.
+func (r *ReconcileMachine) reconcileClusterAnnotations(ctx context.Context, cluster *v1alpha2.Cluster, m *v1alpha2.Machine) error {
+	if !m.DeletionTimestamp.IsZero() || cluster == nil {
+		return nil
+	}
+
+	// If the Machine is a control plane, it has a NodeRef and it's ready, set an annotation on the Cluster.
+	if util.IsControlPlaneMachine(m) && m.Status.NodeRef != nil && m.Status.InfrastructureReady {
+		if _, ok := cluster.Annotations[v1alpha2.ClusterAnnotationControlPlaneReady]; !ok {
```

Comment:
We'll need to nil-check the Annotations map and create it if needed

Reply:
Done, good catch!